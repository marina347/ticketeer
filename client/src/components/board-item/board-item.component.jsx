import React from "react";
import { connect } from "react-redux";
import Tooltip from "@material-ui/core/Tooltip";

import { BoardItemContainer } from "./board-item.styles";
import { selectBoard } from "../../redux/board/board.selectors";

const MAX_LENGTH = 28;

export const BoardItem = ({ history, match, _id, board }) => {
  return (
    <Tooltip
      title={
        <div style={{ fontSize: "16px" }}>
          <p>{board.name}</p>
          <br />
          <p>{board.description}</p>
        </div>
      }
      placement="top"
    >
      <BoardItemContainer
        onClick={() => history.push(`${match.url}/boards/${_id}`)}
      >
        <p id="board_name_text">
          {board.name.length > 20
            ? `${board.name.substring(0, MAX_LENGTH)}...`
            : board.name}
        </p>
      </BoardItemContainer>
    </Tooltip>
  );
};

const mapStateToProps = (state, ownProps) => ({
  board: selectBoard(ownProps._id)(state) || null,
});

export default connect(mapStateToProps)(BoardItem);
