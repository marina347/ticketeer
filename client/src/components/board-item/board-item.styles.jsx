import styled from "styled-components";

export const BoardItemContainer = styled.div`
  display: flex;
  font-size: 22px;
  font-family: "Nunito", sans-serif;
  font-weight: bold;
  background-color: #644cad;
  border: 1px solid grey;
  flex-direction: column;
  padding: 40px;
  cursor: pointer;
  -moz-osx-font-smoothing: grayscale;
  backface-visibility: hidden;
  opacity: 0.8;
  transition: transform 0.25s ease-out;
  overflow: hidden;
  color: gainsboro;

  &:hover {
    transform: scale(1.05);
  }

  @media (max-width: 768px) {
    grid-template-columns: 1fr 1fr;
    padding: 30px;
  }
`;
