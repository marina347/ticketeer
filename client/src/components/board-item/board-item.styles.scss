.board-item {
  font-size: 2.5rem;
  background-color: var(--color-grey-light-1);
  padding: 3rem;

  // flex: 1 0 30%;
  flex: 0 0 48%;

  margin-bottom: 2rem;

  transition: all 0.3s;
  overflow: hidden;
  backface-visibility: hidden;
  border-radius: 10px;
  box-shadow: 0 1rem 1rem rgba(0, 0, 0, 0.15);
  opacity: 0.8;
  cursor: pointer;
  color: var(--color-grey-dark-1);

  &:hover {
    transform: scale(1.02);
    box-shadow: 0 2rem 2rem rgba(0, 0, 0, 0.15);
  }

  &__name {
    text-transform: uppercase;
    font-weight: 600;
    margin-bottom: 1rem;
    background-image: linear-gradient(
      to right,
      var(--color-primary-light),
      var(--color-primary-dark)
    );
    -webkit-background-clip: text;
    color: transparent;
  }

  &__description {
    font-size: 1.4rem;
  }
}
