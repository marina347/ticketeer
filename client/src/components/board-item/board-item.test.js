import React from "react";
import { shallow} from "enzyme";
import { BoardItem } from "./board-item.component";

describe("BoardItem component", () => {
  let wrapper;
  const mockMatch = {
    url: "/home",
  };
  const mockHistory = {
    push: jest.fn(),
  };
  const mockBoard = { _id: "1", name: "Board1" };
  const id = "1";

  beforeEach(() => {
    const mockProps = {
      history: mockHistory,
      match: mockMatch,
      board: mockBoard,
      _id: id,
    };

    wrapper = shallow(<BoardItem {...mockProps} />);
  });

  it("Should render BoardItem component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Should call history.push with the right value when board item clicked", () => {
    wrapper.find('[id="board_item"]').simulate("click");
    expect(mockHistory.push).toHaveBeenCalledWith(
      `${mockMatch.url}/boards/${id}`
    );
  });

  it("Should render the board name as the text", () => {
    const boardNameText = wrapper.find('[id="board_item_name"]').text();
    expect(boardNameText).toBe("Board1");
  });
});
