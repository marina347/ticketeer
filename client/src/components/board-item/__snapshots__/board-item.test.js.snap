// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardItem component Should render BoardItem component 1`] = `
<WithStyles(ForwardRef(Tooltip))
  placement="top"
  title={
    <div
      style={
        Object {
          "fontSize": "16px",
        }
      }
    >
      <p>
        Board1
      </p>
      <br />
      <p />
    </div>
  }
>
  <BoardItemContainer
    onClick={[Function]}
  >
    <p
      id="board_name_text"
    >
      Board1
    </p>
  </BoardItemContainer>
</WithStyles(ForwardRef(Tooltip))>
`;
